### PR TITLE
feature #640: Added endpoint to check permissions

### LINKED ISSUE #640: Implement endpoint to check permissions
Permissions are required to display the components of a specific front page. You should implement an endpoint that will check these permissions, e.g  ```v2/permissions/check```

it will have the following contract:
you receive:
```
{
  permissions: ['users.$userId.selective.get', 'users.$userId.contacts.get' ...],
  values: {
    userId: '93218d08-6dc9-4719-8d96-7c62611a5500',
    ...
  },
}
```
You have to substitute a value in the permissions and check these permissions through a req in the database.
Then you should send an object where the keys will be the same permissions and the values will be boolean

### PR DESCRIPTION
**Pull request by kropli**
- created PermissionController
- created PermissionModule
- implemented checkPermissions method
- described documentation

closes #640

### CODE REVIEW

File: src/v2/api/services/PermissionService.ts [hunk 1]
```diff
@@ -72,4 +77,43 @@ export class PermissionService {
...
     return !!grant?.set;
   }
+
+
+
+  async checkPermissions (userId: string, { permissions, values }: CheckPermissionsDTO) {
+    const user = await this.userRepository.findById(userId);
+    if (user == null) {
+      throw new InvalidEntityIdException('User');
+    }
```
Comment (fokaaas) [inline src/v2/api/services/PermissionService.ts:87]: use pipe instead of this

**Review by fokaaas — CHANGES_REQUESTED** [review]

**Review by fokaaas — APPROVED** [review]

File: src/v2/api/dtos/CheckPermissionsDTO.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+export class CheckPermissionsDTO {
+    @ApiProperty({
+      enum: PERMISSION,
+      type: [PERMISSION],
+    })
+    @IsArray()
+      permissions: PERMISSION[];
+
+    @IsObject()
+    @ApiProperty()
```
Comment (just-hoshinon) [inline src/v2/api/dtos/CheckPermissionsDTO.ts:14]: For example: if a permission will be a `admission.create`, there will be no need to use values field. Thus, this field must be optional in these cases.

File: src/v2/api/controllers/PermissionController.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    name: 'userId',
+    required: true,
+  })
+  @ApiOkResponse({
+    type: CheckPermissionsResponse,
+  })
+  @ApiBadRequestResponse({
+    description: `\n
+    InvalidEntityIdException: 
+      user with such id is not found
```
Comment (just-hoshinon) [inline src/v2/api/controllers/PermissionController.ts:29]: `User`

File: src/v2/api/controllers/PermissionController.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+      
+    DataNotFoundException:
+      Data were not found
+      
+    InvalidBodyException:
+      obj.values: values must be an object
```
Comment (just-hoshinon) [inline src/v2/api/controllers/PermissionController.ts:35]: obj.values is optional

File: src/v2/api/controllers/PermissionController.ts [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+      obj.permissions: permissions must be an array`,
+  })
+  async checkPermissions (
+      @Body() body: CheckPermissionsDTO,
+      @Query('userId', UserByIdPipe) userId: string,
+  ) {
+    return await this.permissionService.checkPermissions(userId, body);
```
Comment (just-hoshinon) [inline src/v2/api/controllers/PermissionController.ts:42]: no await

File: src/v2/api/services/PermissionService.ts [hunk 6]
```diff
@@ -72,4 +74,37 @@ export class PermissionService {
     const grant = this.findGrantMatchesPermission(permission, grants);
     return !!grant?.set;
   }
+
+
+
+  async checkPermissions (userId: string, { permissions, values }: CheckPermissionsDTO) {
```
Comment (just-hoshinon) [inline src/v2/api/services/PermissionService.ts:80]: Now values is optional, pay attention

**Review by just-hoshinon — CHANGES_REQUESTED** [review]

**Review by just-hoshinon — APPROVED** [review]

File: src/v2/api/services/PermissionService.ts [hunk 7]
```diff
@@ -72,4 +74,36 @@ export class PermissionService {
     const grant = this.findGrantMatchesPermission(permission, grants);
     return !!grant?.set;
   }
+
+  async checkPermissions (userId: string, body: CheckPermissionsDTO) {
+    const result = new Map<string, boolean>();
+
+    for (const permission of body.permissions) {
+      const valuesForPermission = body.values? body.values : {};
```
Comment (fokaaas) [inline src/v2/api/services/PermissionService.ts:82]:
```
const valuesForPermission = body.values ?? {};
```

**Review by fokaaas — CHANGES_REQUESTED** [review]

**Review by fokaaas — APPROVED** [review]

**Pull request merged**